### PR TITLE
# 185106775 Seller Two Factor Auth

### PR DESCRIPTION
**Pull request by Jmukakalisa**
**What does this PR do?**

- This PR implements the two factor authentication feature

**Description of Task to be completed?**

- A seller should be able to authenticate with otp code

**How should this be manually tested?**

- Clone this project run the script "npm run dev" then go to ''signin" then signin as a seller ==> "users/verify"

**Any background context you want to provide?**

- N/A

**What are the relevant pivotal tracker/Trello stories?**

- Tracker: 185106775

**Screenshot**

Screenshot 2023-06-01 at 05 45 17

Screenshot 2023-06-01 at 05 45 27

https://github.com/atlp-rwanda/falcons-ec-fe/assets/90443432/34be9bf8-294f-4bb7-aaea-63bd3084b9a2

**Questions**

- None

### CODE REVIEW

File: src/redux/slices/user/twoFactorAuth.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      const response = await axios.post(
+        `${BASE_URL}/users/otp/verify/${data.OTPtoken}`,
+        { otp: data.otp }
+      );
+
+      if (response.status === 200) {
+        toast.success('User authenticated successfully.');       
+      }
+      window.location.href = '/sellerDashboard'; 
+      return response.data;
```
Comment (fab-ryan) [inline src/redux/slices/user/twoFactorAuth.js]: when verification is done you should destruct token from the response and store it as token

**Review by fab-ryan — CHANGES_REQUESTED** [review]
check how your token can be stored without login again

**Review by kirengaboris — APPROVED** [review]

**Review by kirengaboris — APPROVED** [review]

**Review by rwamugema — APPROVED** [review]

File: src/views/twoFactorAuth.jsx [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+          <form className="form" onSubmit={handleSubmit}>
+            <div className="password-input-container">
+              <Input
+                type={showOtp ? 'text' : 'otp'}
+                name="otp"
+                placeholder="Otp Code"
+                value={formData.otp}
+                onChange={handleInputChange}
+                className={otpError ? 'error' : ''}
+              />
```
Comment (fab-ryan) [inline src/views/twoFactorAuth.jsx]: there isn't a property called OTP in Input type as attributes value

**Review by fab-ryan — CHANGES_REQUESTED** [review]
correct that mistake then  I will merge it

**Review by fab-ryan — APPROVED** [review]

**Pull request merged**